bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerFragmentReplicator.java

```diff
@@ -317,7 +318,7 @@ public void writeComplete(int rc, long ledgerId,
                                 ledgerFragmentEntryMcb.processResult(rc, null,
                                         null);
                             }
-                        }, null, BookieProtocol.FLAG_RECOVERY_ADD);
+                        }, null, BookieProtocol.FLAG_RECOVERY_ADD, LedgerType.FORCE_ON_JOURNAL);
```

Comment:
you need to use the type from ledger metadata, no?

Reply:
I think this is a tricky point. The replicator needs to have guarantee that entry is really persisted.
This was a point were I thought that we needed a per entry flag and not a ledger wise flag.
Do you think that replicator should not wait for fsync?

Reply:
@sijie @jvrao in this patch I will use the same LedgerType as it is in LedgerMetadata.
I am evaluating if LedgerFragmentReplicator needs an additional "force()" after the write.
This additional change will be in a further patch, in this patch we do not have force() yet

Reply:
we already have a flag indicating an addEntry operation is for recovery add. every recovery add (both ledger recovery and auto recovery) can (and probably should) happen on journal.